src/shared/parameter_v2/config/examples/Example.yaml

```diff
@@ -0,0 +1,152 @@
+      as ("Stop", "Defense", "Offense"), then this parameter would only accept
+      "Stop", "Defense", and "Offense" as valid values.
+
+      Will only generate for CPP code.
+
+#########################################################################################
+# The following are example of special attributes that can be applied to any parameter
+#########################################################################################
+-
+# Make a parameter value optional (CPP only)
+- bool:
+    name: example_optional_parameter
+    value: false
+    optional: true
+    description: >
+      Optional parameters will generate with std::optional as their value return type.
+      They can accept std::nullopt as a value. If optional is not specified, it is
+      assumed to be false.
+
+      Will only generate for CPP code.
+# Make a parameter a constant value
+- int:
+    name: example_const_parameter
+    min: 0
+    max: 255
+    value: 50
+    const: true
+    description: >
+      Parameters with the const attribute do not generate with an API to change their values.
+      They are constants. If const is not specified, it is assumed to be false.
+
+      Only parameters that are marked as const will be generated for C code. We assume
+      that C code is firmware that we cannot change at runtime because it is running
+      on the robots (not the main AI computer).
+
+#########################################################################################
+# The following are examples of how to include other configs
+#########################################################################################
+# Config files can include other configs. This is how the parameter hierarchy is built
+# up, and allows us to share parameters between configs. For example, this makes it
+# easy to define some passing parameters in one location, and share those parameters
+# with the subset of Plays that need them.
+#
+# All that needs to be specified is the name the config will be reference as, and the
+# path to the file to include. The include paths follow the same concepts at Bazel:
+# - Files in the same directory can be specified with a colon preceding the filename
+#   Eg. include: ":SomeFile.yaml"
+# - Otherwise, the absolute filepath relative to the root of the config directory
+#   (ie. the top-level directory containing all the config files) must be provided.
+#   The name of the file to be included must be the last component of the path, and
+#   separated by a colon.
+#   Eg. include: "Config/examples:Example.yaml"
+#
+# In the generated code, included configs and their attributes can be accessed by
+# the name of the included config.
+# Eg. example_included_config_in_the_same_directory.foo_bool
+
+- config:
+    name: example_included_config_in_the_same_directory
+    path: ":Foo.yaml"
+- config:
+    name: example_included_config_with_absolute_path
+    path: "config/examples:Foo.yaml"
```

Comment:
Would be helpful to see the equivalent C/C++ `struct` corresponding to this.

Reply:
Fair enough, but do you think that's super useful as documentation here? It is technically an implementation detail so we should be somewhat cautious about exposing too much.

Do you mean just see how a user would access the included config?

ie. `config->example->example_included_config_with_absolute_path->foo_bool`?

Reply:
Yeah, technically an implementation detail, but I think we made a lot of changes to the top level config so that it would neatly translate to C/C++, so I think seeing that structure here would help give me confidence that we've come up something that actually does sanely translate.